extensions/ql-vscode/src/mocks/mock-gh-api-server.ts

```diff
@@ -33,8 +49,98 @@ export class MockGitHubApiServer {
     // TODO: Implement logic to list all available scenarios.
   }
 
-  public recordScenario(): void {
-    // TODO: Implement logic to record a new scenario to a directory.
+  public async recordScenario(): Promise<void> {
+    if (this.recorder.isRecording) {
+      void window.showErrorMessage('A scenario is already being recorded. Use the "Save Scenario" or "Cancel Scenario" commands to finish recording.');
+      return;
+    }
+
+    this.recorder.start();
+    await commands.executeCommand('setContext', 'codeQL.mockGitHubApiServer.recording', true);
+
+    await window.showInformationMessage('Recording scenario. To save the scenario, use the "CodeQL Mock GitHub API Server: Save Scenario" command.');
```

Comment:
Does this look better if we add some new lines? I'm a bit surprised it's not multi-line.

<img width="546" alt="image" src="https://user-images.githubusercontent.com/311693/197148805-da8f81ca-6a64-4525-bda2-48eabd8a8304.png">

Reply:
It seems like adding newlines doesn't actually change this since the notification is still collapsed. I'm not sure why it's collapsed, it might be because it doesn't have an action button? We could add a "Cancel" button to it just to make sure it's expanded but that feels a bit hacky.

Reply:
That's a shame. You're right that it feels a bit hacky.